main_test.go

```diff
@@ -11,19 +11,81 @@ import (
 	"go.uber.org/zap"
 )
 
+func Test_NewAuthSite(t *testing.T) {
+	assert := assert.New(t)
+	tests := []struct {
+		name    string
+		value   string
+		wantErr bool
+		want    AuthSite
+		err     string
+	}{
+		{
+			name:    "no protocol",
+			value:   "noprotocol:9000",
+			wantErr: false,
+			want:    AuthSite{"noprotocol:9000", ""},
+		},
+		{
+			name:    "no port",
+			value:   "noport",
+			wantErr: false,
+			want:    AuthSite{"noport:80", ""},
+		},
+		{
+			name:    "ftp",
+			value:   "ftp://ftp:9000",
+			wantErr: false,
+			want:    AuthSite{"ftp:9000", ""},
+		},
+		{
+			name:    "http",
+			value:   "http://http:9000",
+			wantErr: false,
+			want:    AuthSite{"http:9000", ""},
+		},
+		{
+			name:    "with path",
+			value:   "http://withpath:9000/test/path",
+			wantErr: false,
+			want:    AuthSite{"withpath:9000", "/test/path"},
+		},
+		{
+			name:    "http.com",
+			value:   "http://http.com:9000",
+			wantErr: false,
+			want:    AuthSite{"http.com:9000", ""},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := NewAuthSite(tc.value)
+			if tc.wantErr {
+				assert.Error(err)
+				assert.Contains(err.Error(), tc.err)
```

Comment:
It's not much of a difference, but there is assert.ErrorContains()

Reply:
Thanks, this does help it fail tests better (if it fails here).